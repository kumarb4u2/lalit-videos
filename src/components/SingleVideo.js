import React from 'react';
import styled from 'styled-components';

const StyledVideo = styled.video`
  cursor: pointer;
`;

const VideoContainer = styled.div`
  border: 2px solid black;
`;

const Title = styled.span`
  font-weight: 500;
`;

const SingleVideo = ({ src, onClickHandler, title }) => (
  <VideoContainer>
    <StyledVideo
      width="100%"
      title="Click to play"
      onClick={() => {
        onClickHandler(title);
      }}
    >
      <source src={src} type="video/mp4" />
      Your browser does not support the video tag.
    </StyledVideo>
    <Title>{title}</Title>
  </VideoContainer>
);

export default SingleVideo;
