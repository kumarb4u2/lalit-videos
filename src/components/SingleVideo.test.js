import React from 'react';
import { shallow } from 'enzyme';
import SingleVideo from './SingleVideo';

describe('SingleVideo', () => {
  it('should match snapshot in default state', () => {
    const playVideo = jest.fn();
    const component = shallow(
      <SingleVideo
        title="my video"
        src="http://localhost:4000/myvideo.mp4"
        onClickHandler={playVideo}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
