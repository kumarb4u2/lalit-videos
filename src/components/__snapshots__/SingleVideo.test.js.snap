// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleVideo should match snapshot in default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleVideo
    src="http://localhost:4000/myvideo.mp4"
    title="my video"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          onClick={[Function]}
          title="Click to play"
          width="100%"
        >
          <source
            src="http://localhost:4000/myvideo.mp4"
            type="video/mp4"
          />
          Your browser does not support the video tag.
        </ForwardRef>,
        <ForwardRef>
          my video
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <source
              src="http://localhost:4000/myvideo.mp4"
              type="video/mp4"
            />,
            "Your browser does not support the video tag.",
          ],
          "onClick": [Function],
          "title": "Click to play",
          "width": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "http://localhost:4000/myvideo.mp4",
              "type": "video/mp4",
            },
            "ref": null,
            "rendered": null,
            "type": "source",
          },
          "Your browser does not support the video tag.",
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  cursor: pointer;
",
            ],
          },
          "displayName": "styled.video",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "video",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "my video",
        },
        "ref": null,
        "rendered": "my video",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  font-weight: 500;
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  border: 2px solid black;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            onClick={[Function]}
            title="Click to play"
            width="100%"
          >
            <source
              src="http://localhost:4000/myvideo.mp4"
              type="video/mp4"
            />
            Your browser does not support the video tag.
          </ForwardRef>,
          <ForwardRef>
            my video
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <source
                src="http://localhost:4000/myvideo.mp4"
                type="video/mp4"
              />,
              "Your browser does not support the video tag.",
            ],
            "onClick": [Function],
            "title": "Click to play",
            "width": "100%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "http://localhost:4000/myvideo.mp4",
                "type": "video/mp4",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            "Your browser does not support the video tag.",
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
  cursor: pointer;
",
              ],
            },
            "displayName": "styled.video",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "video",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "my video",
          },
          "ref": null,
          "rendered": "my video",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  font-weight: 500;
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  border: 2px solid black;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
